Zend/zend_execute.c

```diff
@@ -828,6 +828,21 @@ ZEND_API bool zend_verify_scalar_type_hint(uint32_t type_mask, zval *arg, bool s
 	return zend_verify_weak_scalar_type_hint(type_mask, arg);
 }
 
+ZEND_COLD zend_never_inline void zend_verify_class_constant_type_error(const zend_class_constant *c, const zval *constant)
+{
+	/* we _may_ land here in case reading already errored and runtime cache thus has not been updated (i.e. it contains a valid but unrelated info) */
```

Comment:
I'm not sure if this can happen here. I think this is used for cases where `zend_verify_property_type` has side effects and throws before this function is executed. This can still actually happen with this implementation but it shouldn't according to the RFC.

Reply:
Yeah, I wasn't sure either, but I hoped that this will fix some of the memory leaks.. I haven't had the chance to get rid of it after it turned out that it didn't fix anything :/